.circleci/dist_compile.yml

```diff
@@ -0,0 +1,697 @@
+      image: ubuntu-2204:2022.10.1 # Ubuntu 22.04
+    resource_class: 2xlarge
+    steps:
+      - checkout
+      - update-submodules
+      - run:
+          name: "Build images for ubuntu-CPP declared in docker-compose.yml"
+          command: docker-compose build ubuntu-cpp
+      - run:
+          name: "Run ubuntu-cpp image"
+          command: docker-compose run -e NUM_THREADS=16 --rm ubuntu-cpp
+
+  docker-compose-centos-cpp:
+    machine: # Use machine executor type to use docker-compose
+      # Ubuntu 22.04 is the base image but we use centos 8 on
+      # the docker-compose container. There are no centos on
+      # circle-ci base images
+      image: ubuntu-2204:2022.10.1
+    resource_class: 2xlarge
+    steps:
+      - checkout
+      - update-submodules
+      - run:
+          name: "Build images for centos-cpp declared in docker-compose.yml"
+          command: docker-compose build centos-cpp
+      - run:
+          name: "Run centos-cpp image"
+          command: docker-compose run -e NUM_THREADS=16 --rm centos-cpp
+
+  docker-compose-presto-native:
+    machine: # Use machine executor type to use docker-compose
+      image: ubuntu-2204:2022.10.1
+    resource_class: 2xlarge
+    environment:
+      PRESTODB_REPOSITORY: https://github.com/prestodb/presto
+    steps:
+      - run:
+          name: "Checkout prestodb"
+          command: git clone git@github.com:prestodb/presto.git
+      - run:
+          name: "Update presto-native to latest Velox"
+          command: |
+            cd presto
+            git submodule update --init --recursive
+            cd presto-native-execution
+            git submodule update --remote --merge
+      - run:
+          name: "Build presto-native container"
+          command: |
+            git config --global user.email "velox@users.noreply.github.com"
+            git config --global user.name "velox"
+            cd presto/presto-native-execution
+            make runtime-container
+            
+
+workflows:
+
+  longer-fuzzer:
+    when: << pipeline.parameters.run-longer-expression-fuzzer >>
+    jobs:
+      - linux-build:
+          matrix:
+            parameters:
+              presto_expression_fuzzer_run_time: [ 3600 ]
+      - linux-build-options
+      - linux-adapters
+      - macos-build:
+          matrix:
+            parameters:
+              os: [ macos-intel ]
+      - format-check
+      - header-check
+
+  shorter-fuzzer:
+    unless: << pipeline.parameters.run-longer-expression-fuzzer >>
+    jobs:
+      - linux-build
+      - linux-build-options
+      - linux-adapters
+      - macos-build:
+          matrix:
+            parameters:
+              os: [ macos-intel ]
+      - format-check
+      - header-check
+
+
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 3 * * *"
+          filters:
+            branches:
+              only:
+                - main
+    jobs:
+      - doc-gen-job
+      - linux-presto-fuzzer-run
```

Comment:
Should nightly also run longer fuzzer test?

Reply:
I think I will just change the nightlys to run for an hour.